src/main/java/com/iexec/worker/sms/SmsService.java

```diff
@@ -47,81 +49,68 @@ public SmsService(SmsClient smsClient,
         this.workerConfigurationService = workerConfigurationService;
     }
 
+    @Retryable(value = FeignException.class)
     public boolean fetchTaskSecrets(ContributionAuthorization contributionAuth) {
-        String hash = HashUtils.concatenateAndHash(contributionAuth.getWorkerWallet(),
-                contributionAuth.getChainTaskId(), contributionAuth.getEnclaveChallenge());
+        String chainTaskId = contributionAuth.getChainTaskId();
+        SmsRequest smsRequest = buildSmsRequest(contributionAuth);
 
-        Sign.SignatureData workerSignature = Sign.signPrefixedMessage(
-                BytesUtils.stringToBytes(hash), credentialsService.getCredentials().getEcKeyPair());
+        SmsSecretResponse smsResponse = smsClient.getTaskSecrets(smsRequest);
```

Comment:
getTaskSecrets() should return taskSecrets :) ?

Reply:
Okey!